ape_solidity/compiler.py

```diff
@@ -163,79 +163,100 @@ def compile(
         self, contract_filepaths: List[Path], base_path: Optional[Path] = None
     ) -> List[ContractType]:
         # TODO: move this to solcx
-        contract_types = []
-        files = []
-        solc_version = None
+        contract_types: List[ContractType] = []
+        files_by_solc_version: Dict[str, Set[Path]] = {}
+        solc_version_by_file_name: Dict[str, str] = {}
```

Comment:
Can these be more specific like this?
```suggestion
        files_by_solc_version: Dict[Version, Set[Path]] = {}
        solc_version_by_file: Dict[Path, Version] = {}
```